### PR TITLE
feat(launch): new perception launch system

### PR DESCRIPTION
**Pull request by aohsato**
## PR Type

- New Feature

## Related Links

- https://github.com/tier4/perception_ecu_container/pull/9
- https://github.com/tier4/perception_ecu_individual_params/pull/3

## Description

Support for multiple camera modes based on the new launch system.

## Review Procedure

Refere to the [README](https://github.com/tier4/perception_ecu_launch/tree/feat/add_samples) in this PR.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            enable_object_recognition = False
+        elif camera_mode == "object_recognition":
+            enable_object_recognition = True
+        else:
+            raise NotImplementedError(camera_mode)
+
+        container_name = create_container_name(camera_name)
+        rois_name = create_rois_name(camera_name)
+
+        config_v4l2_path = create_indivisual_param_path("v4l2_camera.param.yaml")
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:110]:
```suggestion
        config_v4l2_path = create_individual_param_path("v4l2_camera.param.yaml")
```

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        config_info_path = create_indivisual_param_path("camera_info.yaml")
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:111]:
```suggestion
        config_info_path = create_individual_param_path("camera_info.yaml")
```

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        config_trigger_path = create_indivisual_param_path("trigger.param.yaml")
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:112]:
```suggestion
        config_trigger_path = create_individual_param_path("trigger.param.yaml")
```

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    enable_viewer = bool(context.launch_configurations["launch_image_view"])
+
+    try:
+        with open(camera_modes_path, "r") as f:
+            camera_modes = yaml.safe_load(f)
+    except Exception as e:
+        print(e)
+        return
+
+    def create_indivisual_param_path(filename):
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:79]:
```suggestion
    def create_individual_param_path(filename):
```

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        config_basedir_path = PathJoinSubstitution(
+            [FindPackageShare("individual_params"), "config", LaunchConfiguration("vehicle_id")])
+        return PathJoinSubstitution([config_basedir_path, camera_name, filename])
+
+    def create_launch_path(pkgname, filename):
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:84]:
```suggestion
    def create_launch_path(pkg_name, filename):
```
nits

File: perception_ecu_launch/launch/perception_ecu.launch.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        return PathJoinSubstitution([FindPackageShare(pkgname), "launch", filename])
```
Comment (wep21) [inline perception_ecu_launch/launch/perception_ecu.launch.py:85]:
```suggestion
        return PathJoinSubstitution([FindPackageShare(pkg_name), "launch", filename])
```
nits

**Comment by aohsato** [comment]
@wep21 Do you have any input on the current design of this launch system?
I'm wondering it's a little complicated by using `LaunchContext` and `OpaqueFunction`.

File: perception_ecu_launch/launch/perception_ecu.launch.py (hunk 5 above)
Comment (wep21) [reply perception_ecu_launch/launch/perception_ecu.launch.py:84]: `filename` exists as a noun in a dictionary. I think `file_name` is also suitable to unify the style.

Comment (wep21) [reply perception_ecu_launch/launch/perception_ecu.launch.py:84]: I'll leave it to you.

Comment (aohsato) [reply perception_ecu_launch/launch/perception_ecu.launch.py]: good point!
https://github.com/tier4/perception_ecu_launch/pull/3/commits/c6b8f65a0aab96bc1f167464bb34c580973b5a9e

File: perception_ecu_launch/package.xml [hunk 7]
```diff
@@ -9,9 +9,16 @@
...
+  <exec_depend>sensor_trigger</exec_depend>
+  <exec_depend>image_view</exec_depend>
+  <exec_depend>image_transport_plugins</exec_depend>
+  <exec_depend>individual_params</exec_depend>
+  <exec_depend>tensorrt_yolox</exec_depend>
 
-  <exec_depend>perception_ecu_or_launch</exec_depend>
-  <exec_depend>perception_ecu_calib_launch</exec_depend>
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>autoware_lint_common</test_depend>
```
Comment (wep21) [inline perception_ecu_launch/package.xml:21]:
```suggestion
  <test_depend>ament_lint_common</test_depend>
```

File: edge_auto_jetson_launch/config/v4l2_camera.param.yaml
[the remark this answers is not in the document]
Comment (aohsato) [reply edge_auto_jetson_launch/config/v4l2_camera.param.yaml:1]: @manato Thanks for your comments. I added this file as default, as you mentioned standalone.
https://github.com/tier4/perception_ecu_launch/pull/3/files#diff-34fc2fefaf4d1d738baa3e81c6ac6a3bdd2a00f9d42acbeb049556e49f3c75f0R13

File: edge_auto_jetson_launch/launch/v4l2_camera.launch.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<launch>
+  <!-- image topic name to be published -->
+  <arg name="image_topic" default="image_raw" />
+  <!-- frame id to be published -->
+  <arg name="camera_frame_id" default="camera_frame_id" />
+  <!-- path to the yaml file that contains parameters for v4l2_camera node -->
+  <arg name="v4l2_camera_param_path" default="$(find-pkg-share edge_auto_jetson_launch)/config/v4l2_camera.param.yaml" />
```
[the remark this answers is not in the document]
Comment (aohsato) [reply edge_auto_jetson_launch/launch/v4l2_camera.launch.xml:7]: https://github.com/tier4/perception_ecu_launch/pull/3#discussion_r1209381327

File: edge_auto_jetson_launch/config/v4l2_camera.param.yaml
[the remark this answers is not in the document]
Comment (manato) [reply edge_auto_jetson_launch/config/v4l2_camera.param.yaml:1]: @aohsato 
Thank you for your prompt response. I understood your intention. Then, there is no problem as it is.

**Review by manato — APPROVED** [review]
LGTM. All my concerns have gone. Thank you @aohsato for your contribution.

**Pull request merged**